### PR TITLE
BellNumbers.md

### PR DESCRIPTION
**Pull request by shiwanshi**

### CODE REVIEW

**Review by appgurueu — CHANGES_REQUESTED** [review]
Please use Markdown. Yes, Markdown is a superset of HTML, but if possible, Markdown is to be preferred. You should be using MD headings, paragraphs & line breaks rather than HTML tags.

**Review by appgurueu — CHANGES_REQUESTED** [review]
This is still no proper Markdown. Please cut down on the number of blank lines, use headings instead of bold / emphasis, use a code block for the code and perhaps also for the set.

File: en/Dynamic Programming/Bell Numbers.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+2) It is added to all sets of every partition, i.e., k*S(n, k)
+S(n, k) is called Stirling numbers of the second kind
+First few Bell numbers are 1, 1, 2, 5, 15, 52, 203, …. 
+A Simple Method to compute n’th Bell Number is to one by one compute S(n, k) for k = 1 to n and return sum of all computed values. Refer this for computation of S(n, k).
+Below is Dynamic Programming based implementation of the above recursive code using the Stirling number-
+
+
+
+
+```
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:38]:
```suggestion
```java
```

File: en/Dynamic Programming/Bell Numbers.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ 52
+ 
+Time complexity: O(N^2) 
+
+Auxiliary Space: O(N^2) 
+
+### Reference:
+https://en.wikipedia.org/wiki/Bell_number 
+
+https://en.wikipedia.org/wiki/Bell_triangle
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:105]:
```suggestion
### References

- <https://en.wikipedia.org/wiki/Bell_number>
- <https://en.wikipedia.org/wiki/Bell_triangle>
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: en/Dynamic Programming/Bell Numbers.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+ 52
+ 
+Time complexity: O(N^2) 
+
+Auxiliary Space: O(N^2) 
+
+### References
+
+- <https://en.wikipedia.org/wiki/Bell_number>
+- <https://en.wikipedia.org/wiki/Bell_triangle>
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:105]:
```suggestion
- [Bell Number - Wikipedia](https://en.wikipedia.org/wiki/Bell_number)
- [Bell Triangle - Wikipedia](https://en.wikipedia.org/wiki/Bell_triangle)
```

File: en/Dynamic Programming/Bell Numbers.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+## BELL NUMBERS: Number of ways to Partition a Set
+Given a set of n elements, find number of ways of partitioning it. 
+### Examples:
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:3]:
```suggestion
# BELL NUMBERS

## Number of ways to Partition a Set

Given a set of n elements, find number of ways of partitioning it.
## Examples

```

File: en/Dynamic Programming/Bell Numbers.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+Input: n = 3
+
+Output: Number of ways = 5
+Explanation: Let the set be 
+```
+{1, 2, 3} { {1}, {2}, {3} } { {1}, {2, 3} } { {2}, {1, 3} } { {3}, {1, 2} } { {1, 2, 3} } 
+```
+### Recommended practice
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:20]:
```suggestion
```

## Recommended practice

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: en/Dynamic Programming/Bell Numbers.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+Bell(n) = \sum_{k=1}^{n}S(n,k)                 
+Value of S(n, k) can be defined recursively as, S(n+1, k) = k*S(n, k) + S(n, k-1)
+How does above recursive formula work? 
+When we add a (n+1)’th element to k partitions, there are two possibilities. 
+1) It is added as a single element set to existing partitions, i.e, S(n, k-1) 
+2) It is added to all sets of every partition, i.e., k*S(n, k)
+S(n, k) is called Stirling numbers of the second kind
+First few Bell numbers are 1, 1, 2, 5, 15, 52, 203, …. 
+A Simple Method to compute n’th Bell Number is to one by one compute S(n, k) for k = 1 to n and return sum of all computed values. Refer this for computation of S(n, k).
+Below is Dynamic Programming based implementation of the above recursive code using the Stirling number-
```
Comment (Panquesito7) [inline en/Dynamic Programming/Bell Numbers.md:35]: There's missing spaces here. Also, I think you need to end the codeblock before all the text.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**